### ISSUE TITLE
Recipes being cached even when removed

### ISSUE THREAD

**Issue by VariableVixen** [mezz/JustEnoughItems#1583]
Similar to #705, but I didn't see some particulars noted there that I have here:

I moved all recipes for crafting sticks to a different mechanic and used `recipes.remove(<ore:stickWood>)` in my CraftTweaker script, and JEI no longer shows any of the vanilla table stick recipes _except_ for [this one](https://i.imgur.com/IwYtflc.png) which:

- accepts two of any `logWood`, one atop the other
- does NOT show a [recipe ID](https://i.imgur.com/lld1ixC.png)
- cannot actually [be crafted](https://i.imgur.com/jfjN0RR.png) even using the exact recipe (down to the item) shown in that screenshot

All other vanilla workbench recipes to craft sticks have been removed from JEI, as expected. This recipe is, IIRC, not vanilla, but I don't know the originating mod. If I had to make a guess, I'd say that maybe some mod is adding its recipes at a strange time and this is getting stuck in the cache? I have _other_ items that I've removed all vanilla workbench recipes for, and they aren't showing up in JEI like this anymore. As an earlier test, using `recipes.removeAll()` in my script left [over a thousand pages](https://i.imgur.com/SoUBLuX.png) of vanilla workbench recipes showing, none of which seemed to actually be craftable. At that time (when I was clearing _all_ vanilla workbench recipes) my console also got spammed with messages like this one when I looked up recipes, suggesting that maybe JEI or some of its plugins don't expect some recipes to be just wiped out:

```
[Client thread/ERROR] [jei]: Error caught from Recipe Category: mezz.jei.plugins.vanilla.crafting.CraftingRecipeCategory
java.lang.NullPointerException: null
	at com.brandon3055.draconicevolution.integration.jei.EIOSpawnerRecipesWrapper.setRecipe(EIOSpawnerRecipesWrapper.java:52) ~[EIOSpawnerRecipesWrapper.class:?]
	at mezz.jei.plugins.vanilla.crafting.CraftingRecipeCategory.setRecipe(CraftingRecipeCategory.java:90) ~[CraftingRecipeCategory.class:?]
	at mezz.jei.gui.recipes.RecipeLayout.create(RecipeLayout.java:69) [RecipeLayout.class:?]
	at mezz.jei.gui.recipes.RecipeGuiLogic.getRecipeLayouts(RecipeGuiLogic.java:198) [RecipeGuiLogic.class:?]
	at mezz.jei.gui.recipes.RecipesGui.updateLayout(RecipesGui.java:477) [RecipesGui.class:?]
	at mezz.jei.gui.recipes.RecipesGui.onStateChange(RecipesGui.java:523) [RecipesGui.class:?]
	at mezz.jei.gui.recipes.RecipeGuiLogic.previousPage(RecipeGuiLogic.java:264) [RecipeGuiLogic.class:?]
	at mezz.jei.gui.recipes.RecipesGui.handleMouseInput(RecipesGui.java:294) [RecipesGui.class:?]
	at net.minecraft.client.gui.GuiScreen.handleInput(GuiScreen.java:501) [blk.class:?]
	at net.minecraft.client.Minecraft.runTick(Minecraft.java:1759) [bib.class:?]
	at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1098) [bib.class:?]
	at net.minecraft.client.Minecraft.run(Minecraft.java:3941) [bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_211]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_211]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_211]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_211]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_211]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_211]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_211]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_211]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
```

I reported this to the CraftTweaker github too (https://github.com/CraftTweaker/CraftTweaker/issues/684) but this latest update with the stick recipes led me to think that it might be either JEI or something else that's either hooking into JEI or registering recipes oddly, so I'm reporting it here too. Let me know if you need a modlist, but I have 305 loaded so it might not be _entirely_ helpful.

**Comment by mezz** [comment]
Thanks for the report!
I am pretty sure this is an issue with some specific mods' recipes, or with CraftTweaker. Please let me know if one of the devs from the other issue think it could be a JEI bug.

**Comment by VariableVixen** [comment]
I don't think it's a problem with CraftTweaker, only because _most_ recipes I modify with it have the changes reflected just fine in JEI. I've been building a crafting overhaul, and so far thousands of recipes have been removed from the vanilla table without any issue. The one with the sticks was the first problem I found like this. In the linked issue, someone else was having the same or a similar problem, and believes the cause to be EnderIO/AE2 (because they have their recipes loaded from external files and "and overwrite any changes" from CT). I'll stick `recipes.removeAll()` back in and see what mods' items are still showing up in the vanilla table handler pages of JEI, then open some issues on their trackers to see if it's that. Maybe someone's just doing recipes weirdly or something...

My earlier guess of JEI as a potential culprit was because I thought that perhaps it was caching recipes _before_ CT modified them, but given that most recipes are updated without issue, that doesn't really make sense. I'll close this now, and if other devs say it's a JEI thing I'll reopen and link the issues in question.
